src/net/sourceforge/kolmafia/textui/langserver/FilesMonitor.java

```diff
@@ -18,6 +20,44 @@ protected FilesMonitor(final AshLanguageServer parent) {
     this.parent = parent;
   }
 
+  /** Updates the information we have about a file based on information from the client. */
+  public void updateFile(final File file, final String text, final int version) {
+    if (file == null) {
+      return;
+    }
+
+    synchronized (this.parent.scripts) {
+      final Script script = this.getScript(file);
+
+      final List<Script.Handler> handlers = this.findHandlers(file);
+
+      if (handlers.size() > 0
+          && ((text != null && script.version >= version)
+              || (script.text == null && text == null))) {
+        // We already have a working handler using an
+        // up-to-date version of that file.
+        return;
+      }
+
+      if (script.text == null || script.version < version || script.text != null && text == null) {
```

Comment:
`script.text != null` is unnecessary because it would be caught by the first check `script.text == null`.